spec/betterc.dd

```diff
@@ -0,0 +1,48 @@
+Ddoc
+
+$(SPEC_S Better C,
+
+`-betterC` is a Command-Line flag for `dmd`, which generated light-weight,
+barebone, runtime-free binaries with are suitable for linking from C.
+
+---
+void main()
+{
+    import core.stdc.stdio : printf;
+    printf("Hello betterC");
+}
+---
+
+$(CONSOLE
+> dmd -betterC hello.d
+Hello betterC
+)
+
+$(H2 $(LNAME2 effects, Effects))
+
+    $(OL
+        $(LI Doesn't link DRuntime and standard library by default)
+	    $(LI Doesn't generate ModuleInfo entries)
+    )
+
+$(H2 $(LNAME2 consequences, Consequences))
+
+As no DRuntime is available, many D features won't work. For example:
+
+	$(OL
+	    $(LI No static module constructors or deconstructors)
+	    $(LI No Garbage Collection)
+	    $(LI No dynamic arrays and associative arrays)
+	    $(LI No Classes)
```

Comment:
Yeah, TypeInfo is useful but this emits a reference to the base class which is found in druntime.

What I think we should do is to massively simplify the typeinfo so the runtime library defines it. The compile only outputs the bare minimum for classes. So when it sees `class Foo {}`, the compiler basically does

```
class TypeInfo_Foo : TypeInfo_Class {
   override TypeInfo_Class base() const { return whatever_symbol; }
   override immutable(Interface[]) interfaces const() { return whatever_generated_static_data; }
}
```

Something along those lines - the minimum needed for dynamic casting. Let the rest be defined with `rtInfo` and the library-defined base class. I didn't even put the initializer in here since that is static data the compiler generates and can be added via rtInfo template to typeinfo if desired, but you can just use something similar to `__traits(classInstanceSize)` (the actual data doesn't exist via traits now, but it could).


Anyway, much `-betterC` probably wouldn't want to use classes since they do require a runtime module and you don't want to provide your own out of fear of ABI incompatibilities.  But with this scheme, it would be pretty easy to opt in to as little or as much info as you want.